nibabel/openers.py

```diff
@@ -17,11 +17,45 @@
 GZIP_MAX_READ_CHUNK = 100 * 1024 * 1024  # 100Mb
 
 
+class BufferedGzipFile(gzip.GzipFile):
+    """GzipFile able to readinto buffer >= 2**32 bytes."""
+    def __init__(self, fileish, mode='rb', compresslevel=9, buffer_size=2**32-1):
+        super(BufferedGzipFile, self).__init__(fileish, mode=mode, compresslevel=compresslevel)
+        self.buffer_size = buffer_size
+
+        # Speedup for #209; attribute not present in in Python 3.5
+        # open gzip files with faster reads on large files using larger chunks
+        # See https://github.com/nipy/nibabel/pull/210 for discussion
+        if hasattr(self, 'max_chunk_read'):
+            gzip_file.max_read_chunk = GZIP_MAX_READ_CHUNK
+
+    def readinto(self, buf):
+        """Uses self.buffer_size to do a buffered read.
+
+        This works around a known issue in Python 3.5.
+        See https://bugs.python.org/issue25626"""
+        n_bytes = len(buf)
+        try:
+            # This works around a known issue in Python 3.5.
+            # See https://bugs.python.org/issue25626
+            mv = memoryview(buf)
```

Comment:
Point taken about waiting on an OverflowError. What about making the check explicitly for Python 3.5.0, instead of testing for `memoryview`? 3.5.1 should fix the issue, and I'm not super comfortable with using a hand-rolled buffering loop by default on 2.7+ because of a 3.5.0-specific bug.

And my understanding is that `readinto` hasn't been deprecated; it's just been pushed [up the hierarchy](https://docs.python.org/3/library/io.html#io.BufferedIOBase.readinto).